docs/guides/gradle.md

```diff
@@ -0,0 +1,166 @@
+[this Gradle doc](https://docs.gradle.org/current/userguide/dependency_management_for_java_projects.html)
+for more info.
+</p>
+
+### Java web server
+
+If your project contains a JavaScript frontend, you may declare a `web-server` subproject, which
+processes the HTTP requests from JS. In `web-server/build.gradle`:
+```groovy
+dependencies {
+    implementation("io.spine:spine-web:${spine.version()}")
+    implenemtation(project(':users'), project(':trains'))
+}
+```
+The `spine-web` artifact provides the components for handling requests from a JavaScript
+frontend. Note that `web-server` depends on all the server-side subprojects in order to be able
+to dispatch commands to and read states from their respective Bounded Contexts.
+
+<p class="note">
+Note the use of `spine.version()`. This method provides the framework version used by the current
+version of the plugin. Prefer this construction over a hardcoded library version for `spine-money`,
+`spine-web`, etc. 
+</p>
+
+For any specific subproject, you can configure to run or skip certain code generation routines.
+For example:
+```groovy
+spine.enableJava {
+    server() // Enable Server API.
+    codegen {
+        protobuf = true // Generate default Protobuf Java classes.
+        spine = false // Avoid enhancing Protobuf Java classes with Spine validation, interfaces, etc.
+        grpc = true // Generate gRPC stubs and implementation bases from Protobuf service definitions.  
+    }
+}
+```
+
+### JavaScript client
+
+Finally, a JavaScript client is also one or more Gradle subprojects. In `build.gradle` for those
+subprojects, declare:
+```groovy
+spine.enableJavaScript()
+
+dependencies {
+    protobuf(project(':model'))
+}
+```
+This configuration sets up JavaScript code generation from the `model` definitions. Handle NPM
+dependencies separately (e.g. adding the dependency for [`spine-web`](https://www.npmjs.com/package/spine-web)).
+
+## Verbose configuration
+
+If the Bootstrap configuration is not customizable enough for you, there are other Gradle plugins
+which may provide the right API.
+
+Those plugins are Spine Model Compiler for Java subprojects and Spine ProtoJs plugin for JavaScript
+submodules. Under the hood, Bootstrap uses those plugins to do the work. This means that Bootstrap
+automatically applies the correct low-level plugin for you.
+
+### Model Compiler
+
+Spine Model Compiler is a Gradle plugin which executes all the code generation routines via several
+Gradle tasks as well as the `modelCompiler { ... }` extension, which allows you to configure those
+tasks The tasks are:
+
+ - `preClean` — deletes directories generated by Spine if `clean` task is also executed. Append
+   directories to `modelCompiler.dirsToClean` to delete some other dirs which could be created
+   during code generation. 
+ - `mergeDescriptorSet` & `mergeTestDescriptorSet` — generate a merged Protobuf type descriptor set
```

Comment:
If we speak "[task] deletes" in the `preClean` section, we should speak "generates" etc in the following sections as well.